src/Shared/Microsoft.DotNet.Kusto/KustoHelpers.cs

```diff
@@ -162,7 +163,49 @@ public IKustoIngestClient GetClient()
             if (string.IsNullOrWhiteSpace(ingestConnectionString))
                 throw new InvalidOperationException($"Kusto {nameof(_kustoOptions.CurrentValue.IngestConnectionString)} is not configured in settings or related KeyVault");
 
-            return _clients.GetOrAdd(ingestConnectionString, _ => KustoIngestFactory.CreateQueuedIngestClient(ingestConnectionString));
+            return _clients.GetOrAdd(ingestConnectionString, _ =>
+                // Since we will hand this out to multiple callers, it's important we don't let it get disposed.
+                new NonDisposable(KustoIngestFactory.CreateQueuedIngestClient(ingestConnectionString))
```

Comment:
Are we concerned about leaking clients if GetOrAdd calls the callback multiple times?

Reply:
I am slightly?Plus if there is anything that _needs_ to get cleaned up, they should have finalizers, otherwise they'll get GC'd eventually.

But it already "leaks" clients whenever the string changes, so it's probably not a big deal?  I was considering changing it from a dict to just a single cached value that flops if the current value doesn't match. I'm more concerned about the ones we just hold onto forever.

To be really fancy, it should be a ref-counting struct with "current" and "old ones that haven't reached zero yet". But that's... a bit much. :-)

Reply:
Mostly I just wanted it not to throw an ObjectDisposedException.

Right now no one ever disposes them, so it's not a problem. But Missy almost did, and I realized it would be really bad. :-)

Reply:
I checked with Chad before I made the assumption that the Kusto client would just "handle everything" for me, and that's what spurred this change :)